import.py

```diff
@@ -649,7 +653,77 @@ def write_to_file(cont: str, filename: str) -> None:
         f.write(cont)
 
 
+def download_decompme(parsed_url: urllib.parse.ParseResult) -> None:
+
+    try:
+        # unique decomp.me identifier for scratches
+        slug = parsed_url.path[1:].split("/")[1]
+    except Exception:
+        print(traceback.format_exc())
+        print(
+            "Failed to parse decomp.me url, it should look like:  https://decomp.me/scratch/aBc12"
+        )
+
+    try:
+
+        response_str = urllib.request.urlopen(f"https://decomp.me/api/scratch/{slug}")
+        response_json = json.load(response_str)
+        func_name = response_json["name"]
+    except Exception:
+        print(traceback.format_exc())
+        print("Failed to query function information from decomp.me")
+
+    dirname = create_directory(func_name)
+
+    try:
+        content = urllib.request.urlopen(f"https://decomp.me/api/scratch/{slug}/export")
+        zip = zipfile.ZipFile(BytesIO(content.read()))
+        zip.extractall(dirname)
+    except Exception:
+        print(traceback.format_exc())
+        print("Failed to download function data from decomp.me")
+
+    try:
+        os.rename(f"{dirname}/code.c", f"{dirname}/base.c")
+        os.rename(f"{dirname}/ctx.c", f"{dirname}/ctx.h")
+        if not os.path.exists("compile.sh"):
+            print(
+                f"Warning!! 'compile.sh' not found in current directory, so it will not be copied to the prepared folder, {dirname}, and this is required by the permuter"
+            )
+        else:
+            shutil.copyfile("compile.sh", f"{dirname}/compile.sh")
+            os.chmod(
+                f"{dirname}/compile.sh",
+                os.stat(f"{dirname}/compile.sh").st_mode | stat.S_IEXEC,
+            )
+
+        with open(f"{dirname}/base.c", "r") as f:
+            filedata = f.read()
+        with open(f"{dirname}/base.c", "w") as f:
+            f.write('#include "ctx.h"\n')
+            f.write(filedata)
+
+        print(
+            f"Success!! Prepared new folder, {dirname}, that is ready for the permuter script! Run with 'python3 permuter.py {dirname} [additional_flags]'"
+        )
+
+    except Exception:
+        print(traceback.format_exc())
+        print(
+            "Exception occured when making final preparations to files in the new directory for the permutor"
+        )
+
+
 def main() -> None:
+
+    api_base = os.environ.get("DECOMPME_API_BASE", "https://decomp.me")
```

Comment:
API_BASE is the path to the Python backend, which can run on a separate port from the backend when running decomp.me locally. We should be testing against the frontend path here, so split this into DECOMPME_BASE and DECOMPME_API_BASE. And make them global constants and use them in download_decompme too

Reply:
Should I use "/scratch/" url path param in the base string, or keep it out?

Reply:
keep it out

Reply:
nevermind, I see it's used like this:
```f"{api_base}/api/compilers"```

so therefore I am just going to leave it as the base string even without the /api/

Reply:
I made these changes, and went through and changed the references to use these global variables..

There were some places, line in print lines where I thought it made sense to print "decomp.me" instead of the api base... so I left as decomp.me... but let me know if there are any other spots you think it should be changed